fdbserver/workloads/Throttling.actor.cpp

```diff
@@ -0,0 +1,234 @@
+				traceTLogQueue.detail(format("TLog/%s", ss.first.toString().c_str()), ss.second);
+			}
+
+			TraceEvent traceCpuUsage("CpuUsage");
+			for (const auto& ss : healthMetrics.cpuUsage) {
+				self->detailedWorstCpuUsage = std::max(self->detailedWorstCpuUsage, ss.second);
+				traceCpuUsage.detail(format("Storage/%s", ss.first.toString().c_str()), ss.second);
+			}
+
+			TraceEvent traceDiskUsage("DiskUsage");
+			for (const auto& ss : healthMetrics.diskUsage) {
+				self->detailedWorstDiskUsage = std::max(self->detailedWorstDiskUsage, ss.second);
+				traceDiskUsage.detail(format("Storage/%s", ss.first.toString().c_str()), ss.second);
+			}
+		}
+	}
+
+	static Value getRandomValue() { return Standalone<StringRef>(format("Value/%d", g_random->randomInt(0, 10e6))); }
+
+	ACTOR static Future<Void> clientActor(Database cx, ThrottlingWorkload* self) {
+		state ReadYourWritesTransaction tr(cx);
+
+		loop {
+			wait(TokenBucket::startTransaction(&self->tokenBucket));
+			tr.reset();
+			try {
+				state int i;
+				for (i = 0; i < self->readsPerTransaction; ++i) {
+					state Optional<Value> value = wait(tr.get(self->getRandomKey()));
+				}
+				for (i = 0; i < self->writesPerTransaction; ++i) {
+					tr.set(self->getRandomKey(), getRandomValue());
+				}
+				wait(tr.commit());
+				if (g_random->randomInt(0, 1000) == 0) TraceEvent("TransactionCommittedx1000");
+				++self->transactionsCommitted;
+			} catch (Error& e) {
+				// ignore failing transactions
+			}
+		}
+	}
+
+	ACTOR static Future<Void> _setup(Database cx, ThrottlingWorkload* self) {
+		Standalone<StringRef> value(format("%d", self->sendDetailedHealthMetrics ? 1 : 0));
+		setNetworkOption(FDBNetworkOptions::SEND_DETAILED_HEALTH_METRICS, Optional<StringRef>(value));
+		if (!self->sendDetailedHealthMetrics) {
+			// Clear detailed health metrics that are already populated
+			wait(delay(2 * CLIENT_KNOBS->UPDATE_DETAILED_HEALTH_METRICS_INTERVAL));
+			cx->healthMetrics.cpuUsage.clear();
+			cx->healthMetrics.storageQueue.clear();
+			cx->healthMetrics.storageNDV.clear();
+			cx->healthMetrics.tLogQueue.clear();
+		}
+		return Void();
+	}
+
+	ACTOR static Future<Void> _start(Database cx, ThrottlingWorkload* self) {
+		state Future<Void> hmChecker = timeout(healthMetricsChecker(cx, self), self->testDuration, Void());
+
+		state vector<Future<Void>> clientActors;
+		state int actorId;
+		for (actorId = 0; actorId < self->actorsPerClient; ++actorId) {
+			clientActors.push_back(timeout(clientActor(cx, self), self->testDuration, Void()));
+		}
+		wait(hmChecker);
+		return Void();
+	}
+
+	virtual std::string description() { return "Throttling"; }
+	virtual Future<Void> setup(Database const& cx) { return _setup(cx, this); }
+	virtual Future<Void> start(Database const& cx) { return _start(cx, this); }
+	virtual Future<bool> check(Database const& cx) {
+		if (worstStorageQueue == 0 || worstStorageNDV == 0 || worstTLogQueue == 0 || transactionsCommitted == 0)
```

Comment:
This test will not fail if the client stop receiving metrics at some point during the test. This was hiding the bug where onMasterProxiesChanged() was not waited on.

In addition please add a SevError trace event with more detailed information about what is missing